src/Mod/Part/App/Attacher.cpp

```diff
@@ -1700,253 +2003,301 @@ AttachEngineLine *AttachEngineLine::copy() const
-            if (shapes[0]->IsNull() || shapes[1]->IsNull())
-                throw Base::ValueError("Null shape in AttachEngineLine::calculateAttachedPlacement()!");
-
-            const TopoDS_Face &face1 = TopoDS::Face(*(shapes[0]));
-            const TopoDS_Face &face2 = TopoDS::Face(*(shapes[1]));
-
-            Handle(Geom_Surface) hSurf1 = BRep_Tool::Surface(face1);
-            Handle(Geom_Surface) hSurf2 = BRep_Tool::Surface(face2);
-            GeomAPI_IntSS intersector(hSurf1, hSurf2, Precision::Confusion());
-            if (!intersector.IsDone())
-                throw Base::ValueError("AttachEngineLine::calculateAttachedPlacement: Intersection failed");
-
-            const Standard_Integer intLines = intersector.NbLines();
-            if (intLines == 0)
-                throw Base::ValueError("AttachEngineLine::calculateAttachedPlacement: The two shapes don't intersect");
-            if (intLines != 1)
-                throw Base::ValueError("AttachEngineLine::calculateAttachedPlacement: Intersection is not a single curve");
-
-            GeomAdaptor_Curve adapt(intersector.Line(1));
-            if (adapt.GetType() != GeomAbs_Line)
-                throw Base::ValueError("AttachEngineLine::calculateAttachedPlacement: Intersection is not a straight line");
-
-            LineBasePoint = adapt.Line().Location();
-            LineDir = adapt.Line().Direction();
-        }break;
-        case mm1Proximity:{
-            if (shapes.size() < 2)
-                throw Base::ValueError("AttachEngineLine::calculateAttachedPlacement: Proximity mode requires two shapes; only one is supplied");
-            if (shapes[0]->IsNull())
-                throw Base::ValueError("Null shape in AttachEngineLine::calculateAttachedPlacement()!");
-            if (shapes[1]->IsNull())
-                throw Base::ValueError("Null shape in AttachEngineLine::calculateAttachedPlacement()!");
-            BRepExtrema_DistShapeShape distancer (*(shapes[0]), *(shapes[1]));
-            if (!distancer.IsDone())
-                throw Base::ValueError("AttachEngineLine::calculateAttachedPlacement: proximity calculation failed.");
-            if (distancer.NbSolution()>1)
-                Base::Console().Warning("AttachEngineLine::calculateAttachedPlacement: proximity calculation gave %i solutions, ambiguous.\n",int(distancer.NbSolution()));
-            gp_Pnt p1 = distancer.PointOnShape1(1);
-            gp_Pnt p2 = distancer.PointOnShape2(1);
-            LineBasePoint = p1;
-            gp_Vec dist = gp_Vec(p1,p2);
-            if (dist.Magnitude() < Precision::Confusion())
-                throw Base::ValueError("AttachEngineLine::calculateAttachedPlacement: can't make proximity line, because shapes touch or intersect");
-            LineDir = gp_Dir(dist);
-        }break;
-        default:
-            throwWrongMode(mmode);
+                throwWrongMode(mmode);
```

Comment:
What has happened to the `mm1Intersection` case that @LemonBoy added last month?

Reply:
Git rebase hates me, and maybe github hates you?  I swore you made this comment before and I saw it on my first scan, but when I was fixing comments I couldn't find it after looking several times the next day.  In any case, a manual restoration puts it back.  Good catch.  <later>  Oh, I get it, rebase thinks it was an intentional deletion, not a case of code ported from an older version.  Duh!